### PR TITLE
Add documentation for missing settings

### PR DESCRIPTION
**Pull request by oh2fih**
Many settings read from environment variables has been added to the `config.py`, but the documentation is at least 19 months behind. This PR tries to add at least some kind of documentation for all the settings available.

The `README.rst` does not mention these settings at all, but only refers to the defaults. This PR adds a note that most of these defaults including the configuration directory itself can be overwritten by these settings.

### CODE REVIEW

File: docs/general/settings.rst [hunk 1]
```diff
@@ -1,11 +1,163 @@
...
+.. confval:: GELF_SYSLOG_ADDITIONAL_FIELDS
+
+   See https://github.com/keeprocking/pygelf?tab=readme-ov-file#static-fields
+
+Redis
+*****
+
+.. confval:: REDIS_URL
+
+   Url to be used for contacting redis cache.
```
Comment (oh2fih) [inline docs/general/settings.rst:163]: While this section could go to the database section as Redis can be used as a database, for CveXplore Redis is more of a cache, and Redis related parameters are also documented as a part of the Celery backend documentation.

The main argument to have this here is that the 'common' settings for databases won't apply for Redis.

**Review by P-T-I — APPROVED** [review]

**Comment by P-T-I** [comment]
@oh2fih Thanks again for your contribution; I'm hoping to be more "active" in the near future on this project! Nevertheless how would you feel if I would elevate your rights on this repo?

**Comment by oh2fih** [comment]
> Nevertheless how would you feel if I would elevate your rights on this repo?

That would indeed add more flexibility. 😎

**Comment by P-T-I** [comment]
Then I'll do so later this day! Thanks again! 👍

**Pull request merged**